File: src/main/scala/co/gywb/demo/web_front/scala/Hello.scala
package co.gywb.demo.web_front.scala

import javax.ws.rs.core.MediaType
import javax.ws.rs.{GET, Path, PathParam, Produces};

@Path("/hello/scala")
class Hello {
  @GET
  @Path("/{name}")
  @Produces(Array(MediaType.TEXT_PLAIN))
  def greet(@PathParam("name") name: String) = {
    s"Hello, $name, from Scala!";
  }
}

File: src/main/kotlin/co/gywb/demo/web_front/kotlin/Hello.kt
package co.gywb.demo.web_front.kotlin

import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/hello/kotlin")
class Hello {

    @GET
    @Path("/{name}")
    @Produces(MediaType.TEXT_PLAIN)
    fun greet(@PathParam("name") name: String) =
            "Hello, $name, from Kotlin!"

}
